tests/e2e/specs/devconsole-intergration/DevConsoleIntegration.spec.ts

```diff
@@ -0,0 +1,99 @@
+
+const projectAndFileTests: ProjectAndFileTests = e2eContainer.get(CLASSES.ProjectAndFileTests);
+const loginTests: LoginTests = e2eContainer.get(CLASSES.LoginTests);
+const workspaceHandlingTests: WorkspaceHandlingTests = e2eContainer.get(CLASSES.WorkspaceHandlingTests);
+const ocpMainPage: OcpMainPage = e2eContainer.get(CLASSES.OcpMainPage);
+let ocpImportPage: OcpImportFromGitPage;
+let ocpApplicationPage: OcpApplicationPage;
+const kubernetesCommandLineToolsExecutor: KubernetesCommandLineToolsExecutor = new KubernetesCommandLineToolsExecutor();
+
+// works only with no-admin user
+suite(`DevConsole Integration`, async function (): Promise<void> {
+    // test specific data
+    const gitImportRepo: string = 'https://github.com/musienko-maxim/summit-lab-spring-music';
+    const gitImportReference: string = 'pipeline';
+    const projectLabel: string = 'app.openshift.io/runtime=spring';
+    const projectName: string = 'devconsole-integration-test';
+
+    suiteSetup('Create new empty project using ocp', async function (): Promise<void> {
+        kubernetesCommandLineToolsExecutor.loginToOcp();
+        kubernetesCommandLineToolsExecutor.deleteProject(projectName);
+        kubernetesCommandLineToolsExecutor.createProject(projectName);
+    });
+
+    loginTests.loginIntoOcpConsole();
+
+    test('Select test project on DevConsole', async function (): Promise<void> {
+        await ocpMainPage.selectProject(projectName);
+    });
+
+    test('Open import from git project page as Developer', async function (): Promise<void> {
+        ocpImportPage = await ocpMainPage.openImportFromGitPage();
+    });
+
+    test('Fill and submit import data', async function (): Promise<void> {
+        ocpApplicationPage = await ocpImportPage.fitAndSubmitConfiguration(gitImportRepo, gitImportReference, projectLabel);
+    });
+
+    test('Wait until application creates', async function (): Promise<void> {
+        await ocpApplicationPage.waitApplicationIcon();
+    });
+
+    test('Check if application has worked link "Open Source Code"', async function (): Promise<void> {
+        await ocpApplicationPage.waitAndOpenEditSourceCodeIcon();
+    });
+
+    loginTests.loginIntoChe();
+
+    workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
+
+    test('Registering the running workspace', async function (): Promise<void> {
+        registerRunningWorkspace(WorkspaceHandlingTests.getWorkspaceName());
+    });
+
+    test('Check if application source code opens in workspace', async function (): Promise<void> {
+        await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+    });
+
+    test('Check if project and files imported', async function (): Promise<void> {
+        const applicationSourceProjectName: string = GitUtil.getProjectNameFromGitUrl(gitImportRepo);
+        const projectSection: ViewSection = await new SideBarView().getContent().getSection(applicationSourceProjectName as unknown as string);
```

Comment:
So that `applicationSourceProjectName` is already a **string**, then casting it to **unknown** and then back to **string** with `as unknown as string` is unnecessary and redundant.

Would you consider:
```
const projectSection: ViewSection = await new SideBarView().getContent().getSection(applicationSourceProjectName);
```
?

Reply:
true

Reply:
looks better now, thanks